source/LiteralModel.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <mariana-trench/EventLogger.h>
+#include <mariana-trench/LiteralModel.h>
+#include <mariana-trench/Positions.h>
+
+namespace marianatrench {
+
+namespace {
+
+class LiteralModelConsistencyError {
+ public:
+  static void raise(const std::string& what) {
+    ERROR(1, "Literal Model Consistency Error: {}", what);
+    EventLogger::log_event("regex_model_consistency_error", what);
+  }
+};
+
+}
+
+static void check_taint_config_consistency(const std::string& pattern, const TaintConfig& frame) {
+  if (frame.kind() == nullptr) {
+    LiteralModelConsistencyError::raise(fmt::format(
+        "Model for regex `{}` must have a source kind.", pattern));
+  }
+}
+
+static Taint create_taint(const std::string& pattern, const std::vector<TaintConfig>& sources) {
+  Taint all_sources;
+  for (const auto& source : sources) {
+    mt_assert(source.is_leaf());
+    check_taint_config_consistency(pattern, source);
+    all_sources.join_with(Taint{std::move(source)});
+  }
+  return all_sources;
+}
+
+LiteralModel::LiteralModel() : regex_(pattern_) { }
```

Comment:
so we create a `LiteralModel` with an empty string and a pattern for the empty string?
I would prefer if this was more clear in the internal state, for instance we could use a `std::optional<re2::RE2>`.

Reply:
I now used an optional, and we do not match anything if default.